src/sources/kubernetes/mod.rs

```diff
@@ -300,6 +268,93 @@ fn transform_file() -> crate::Result<Box<dyn Transform>> {
     })
 }
 
+/// Contains several regexes that can parse common forms of pod_uid.
+/// On the first message, regexes are tried out one after the other until
+/// first succesfull one has been found. After that that regex will be
+/// always used.
+///
+/// If nothing succeeds the message is still passed.
+fn transform_pod_uid() -> crate::Result<ApplicableTransform> {
+    let mut regexes = Vec::new();
+
+    let namespace_regex = r"(?P<pod_namespace>[0-9a-z.\-]*)";
```

Comment:
should we add this as a config item users can set if somehow their pod_uid is in a different format?

Reply:
I was thinking the same, but:
 1. Such situations should be rare and mostly concern older versions of Kubernetes. Now up to which version is something that we will have to determine.
 2. Following comit will change transforms to be more in line with best effort guarantee, and with that if `pod_uid` has unknown format, event would still be emitted. So users can parse it out with `regex_parser` transform, or any other that they need.